test/integration/connect/envoy/run-tests.sh

```diff
@@ -195,80 +199,41 @@ function run_tests {
   docker cp workdir/. envoy_workdir_1:/workdir
 
   start_consul primary
-  if [ $? -ne 0 ]
-  then
-    capture_logs
-    return 1
-  fi
 
-  if is_set $REQUIRE_SECONDARY
-  then
+  if is_set $REQUIRE_SECONDARY; then
     start_consul secondary
-    if [ $? -ne 0 ]
-    then
-      capture_logs
-      return 1
-    fi
   fi
 
   echo "Setting up the primary datacenter"
   pre_service_setup primary
-  if [ $? -ne 0 ]
-  then
-    echo "Setting up the primary datacenter failed"
-    capture_logs
-    return 1
-  fi
 
-  if is_set $REQUIRE_SECONDARY
-  then
+  if is_set $REQUIRE_SECONDARY; then
     echo "Setting up the secondary datacenter"
     pre_service_setup secondary
-    if [ $? -ne 0 ]
-    then
-      echo "Setting up the secondary datacenter failed"
-      capture_logs
-      return 1
-    fi
   fi
 
   echo "Starting services"
   start_services
-  if [ $? -ne 0 ]
-  then
-    capture_logs
-    return 1
-  fi
 
   # Run the verify container and report on the output
   verify primary
-  TESTRESULT=$?
 
-  if is_set $REQUIRE_SECONDARY && test "$TESTRESULT" -eq 0
-  then
+  if is_set $REQUIRE_SECONDARY; then
     verify secondary
-    SECONDARYRESULT=$?
-
-    if [ "$SECONDARYRESULT" -ne 0 ]
-    then
-      TESTRESULT=$SECONDARYRESULT
-    fi
-  fi
-
-  if [ "$TESTRESULT" -ne 0 ]
-  then
-    capture_logs
   fi
+}
 
-  stop_services primary
+function test_teardown {
+    # Set a log dir to prevent docker-compose warning about unset var
+    export LOG_DIR="workdir/logs/"
 
-  if is_set $REQUIRE_SECONDARY
-  then
-    stop_services secondary
-  fi
+    init_vars
 
+    stop_services primary
```

Comment:
`stop_services` only stops `$REQUIRED_SERVICES`. I don't understand why that is, but changing it appeared to cause some tests to fail. I haven't looked into it much.

Reply:
Oh, I guess it was because of the `-v` removing volumes. The errors were failures to `docker cp` I believe.